src/silverlabnwb/nwb_file.py

```diff
@@ -220,6 +221,10 @@ def add_core_metadata(self):
         # even though it is still stored under /general/stimulus
         if 'stimulus' in self.experiment:
             self.add_general_info('stimulus_notes', self.experiment['stimulus'])
+        # If both metadata and header have a non-empty subject entry, the header gets priority.
+        # If neither have an entry, pynwb defaults to not writing a subject field in the file.
+        if self.subject:
+            self.add_subject(self.subject)
```

Comment:
Are we sure that this way prioritises the header over the metadata? If they both have subject information, does the second call to `add_subject` not do anything, or does it fail (or overwrite it with the second argument)?